lib/src/merge.rs

```diff
@@ -645,6 +645,24 @@ where
         Merge { values }
     }
 
+    /// Creates a new merge by setting the executable bit from the given merge.
+    /// In other words, the file ids from `self` will be preserved.
+    pub fn with_new_file_executable_bit(&self, executable: bool) -> Merge<Option<TreeValue>> {
```

Comment:
nit: Since this is very specific to the current builtin editor, I would move this function to merge_tools/builtin.rs. If the builtin editor had a proper support for conflicts, the resolution might be handled differently.

BTW, `jj file chmod` can be a potential user of this function, but the implementation is relatively simple to just inline.

Reply:
I've moved the helper to `merge_tools/builtin.rs`. I found it quite natural to want such a function right next to the complementary `with_new_file_ids()`.

> If the builtin editor had a proper support for conflicts, the resolution might be handled differently.

Not quite sure how that might look like. I thought this was a pretty reasonable operation to have. Like you said, it's more or less `file chmod`.